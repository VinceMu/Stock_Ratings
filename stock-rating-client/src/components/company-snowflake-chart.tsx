import { ResponsiveContainer, RadarChart, PolarGrid, PolarAngleAxis, PolarRadiusAxis, Radar } from "recharts";

interface CompanySnowflakeChartProps {
    value: number;
    future: number;
    past: number;
    health: number;
    dividend: number;
}

const MAX_SCORE = 10;

export const CompanySnowflakeChart = ({ value, future, past, health, dividend }: CompanySnowflakeChartProps) => {
    const snowflakeData = [
        {
            type: "value",
            score: value
        },
        {
            type: "future",
            score: future
        },
        {
            type: "past",
            score: past
        },
        {
            type: "health",
            score: health
        },
        {
            type: "dividend",
            score: dividend
        }

    ]
    return (
        <ResponsiveContainer width="100%" height="100%">
            <RadarChart cx="50%" cy="50%" outerRadius="80%" data={snowflakeData}>
                <PolarGrid />
                <PolarAngleAxis dataKey="type" />
                <PolarRadiusAxis domain={[0, MAX_SCORE]} />
                <Radar name="CompanyScores" dataKey="score" stroke="#8884d8" fill="#0E6EB8" fillOpacity={0.6} />
            </RadarChart>
        </ResponsiveContainer>
    );

}