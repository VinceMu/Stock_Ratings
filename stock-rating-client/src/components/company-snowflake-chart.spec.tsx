import 'react';
import renderer from 'react-test-renderer';
import { CompanySnowflakeChart } from './company-snowflake-chart';

const data = {
    value: 1,
    future: 2,
    past: 3,
    health: 4,
    dividend: 5
}

test('Company snowflake chart matches snapshot', () => {
    const tree = renderer.create(
        <div style={{ height: "100px", width: "200px" }}>
            <CompanySnowflakeChart
                value={data.value}
                future={data.future}
                past={data.past}
                health={data.health}
                dividend={data.dividend}
            />
        </div >
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
