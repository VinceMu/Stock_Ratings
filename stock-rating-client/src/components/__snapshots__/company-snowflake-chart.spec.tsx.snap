// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company snowflake chart matches snapshot 1`] = `
<div
  style={
    Object {
      "height": "100px",
      "width": "200px",
    }
  }
>
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />
</div>
`;
